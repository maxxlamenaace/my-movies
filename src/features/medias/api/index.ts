import { getMediaCredits } from './get-media-credits/get-media-credits';
import { getMediaDetails } from './get-media-details/get-media-details';
import { getMediaGenres } from './get-media-genres/get-media-genres';
import { getMediaImages } from './get-media-images/get-media-images';
import { getMediaRecommendations } from './get-media-recommendations/get-media-recommendations';
import { getMediaVideos } from './get-media-videos/get-media-videos';
import { getMediasList } from './get-medias-list/get-medias-list';

export {
  getMediaCredits,
  getMediaDetails,
  getMediaGenres,
  getMediaImages,
  getMediaRecommendations,
  getMediaVideos,
  getMediasList,
};
