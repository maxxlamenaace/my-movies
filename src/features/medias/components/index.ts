import CircularRate from './circular-rate/CircularRate';
import MediasAutoSwiper from './medias-auto-swiper/MediasAutoSwiper';
import MediasList from './medias-list/MediasList';
import MediaItem from './media-item/MediaItem';
import MediaBackgroundHeader from './media-background-header/MediaBackgroundHeader';
import MediaInfos from './media-infos/MediaInfos';
import CastSlide from './cast-slide/CastSlide';
import MediaVideosSlide from './media-videos-slide/MediaVideosSlide';
import MediaVideoItem from './media-video-item/MediaVideoItem';
import MediaImagesSlide from './media-images-slide/MediaImagesSlide';
import MediasGrid from './medias-grid/MediasGrid';
import FavoriteItem from './favorite-item/FavoriteItem';
import FavoritesGrid from './favorites-grid/FavoritesGrid';

export {
  CircularRate,
  MediasAutoSwiper,
  MediasList,
  MediaItem,
  MediaBackgroundHeader,
  MediaInfos,
  CastSlide,
  MediaVideosSlide,
  MediaVideoItem,
  MediaImagesSlide,
  MediasGrid,
  FavoriteItem,
  FavoritesGrid,
};
