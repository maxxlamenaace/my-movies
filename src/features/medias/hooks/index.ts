import { useMediaDetails } from './use-media-details/use-media-details';
import { useMediaGenres } from './use-media-genres/use-media-genres';
import { useAllMedias } from './use-all-medias/use-all-medias';
import { useMediasList } from './use-medias-list/use-medias-list';
import { useFavoriteMedias } from './use-favorite-medias/use-favorite-medias';

export { useMediaDetails, useAllMedias, useMediaGenres, useMediasList, useFavoriteMedias };
