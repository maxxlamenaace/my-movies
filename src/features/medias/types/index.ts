export type MediaType = 'movie' | 'tv';
export type MediaCategory = 'popular' | 'top_rated';
export type MediaImageType = 'poster' | 'backdrop';

export type Media = {
  id: number;
  poster_path?: string;
  adult: boolean;
  overview: string;
  genre_ids: number[];
  original_title: string;
  original_language: string;
  media_type?: MediaType;
  title?: string; // Only for movies
  name?: string; // Only for series
  backdrop_path?: string;
  popularity: number;
  vote_count: number;
  video: boolean;
  vote_average: number;
  release_date?: string; // Only for movies
  first_air_date?: string; // Only for series
};

export type MediaDetails = {
  id: number;
  budget: number;
  genres: Genre[];
  homepage: string;
  revenue: number;
  runtime: number;
  status: string;
  tagline: string;
} & Media;

export type MediaCredit = {
  id: number;
  adult: boolean;
  gender: number;
  known_for_department: string;
  name: string;
  original_name: string;
  popularity: number;
  profile_path: string;
  cast_id: number;
  character: string;
  credit_id: string;
  order: number;
};

export type MediaVideo = {
  id: number;
  type: string;
  official: boolean;
  name: string;
  site: string;
  size: number;
  published_at: Date;
  key: string;
};

export type MediaImage = {
  aspect_ratio: number;
  file_path: string;
  height: number;
  width: number;
};

/* export type ProductionCompany = {
  id: number;
  logo_path: string;
  name: string;
  origin_country: string;
};

export type ProductionCountry = {
  iso_3166_1: string;
  name: string;
};

export type SpokenLanguage = {
  iso_639_1: string;
  name: string;
}; */

export type Genre = {
  id: number;
  name: string;
};
