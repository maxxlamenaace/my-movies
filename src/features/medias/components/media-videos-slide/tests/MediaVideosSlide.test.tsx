import React from 'react';
import { cleanup, render } from '@testing-library/react';

import { MediaVideosSlide } from '@/features/medias';
import { mediaVideo } from '@/tests/factories';

describe('MediaVideosSlide', () => {
  afterEach(() => {
    cleanup();
  });

  it('renders correctly', () => {
    const { asFragment } = render(<MediaVideosSlide videos={[mediaVideo()]} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with as many media video item as needed', () => {
    const mediaVideoToBeRendered = mediaVideo();
    const { container } = render(<MediaVideosSlide videos={[mediaVideoToBeRendered]} />);

    expect(container.querySelectorAll('#media-video-item')).toHaveLength(1);
  });
});
