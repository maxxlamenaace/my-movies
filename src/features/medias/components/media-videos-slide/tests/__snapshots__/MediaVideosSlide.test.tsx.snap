// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaVideosSlide renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-10mh0q4"
  >
    <div
      class="swiper swiper-initialized swiper-horizontal swiper-pointer-events"
      id="paginator-swiper"
      style="width: 100%;"
    >
      <div
        class="swiper-wrapper"
        style="cursor: grab; transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
      >
        <div
          class="swiper-slide"
        >
          <div
            class="MuiBox-root css-14x27ji"
            id="media-video-item"
          >
            <iframe
              height="0px"
              src="https://www.youtube.com/embed/key?controls=0"
              style="border: 0px;"
              title="266331283"
              width="100%"
            />
          </div>
        </div>
      </div>
      <div
        class="swiper-button-prev"
      />
      <div
        class="swiper-button-next"
      />
      <div
        class="swiper-pagination swiper-pagination-clickable swiper-pagination-bullets swiper-pagination-horizontal"
      />
    </div>
  </div>
</DocumentFragment>
`;
